src/client/mod.rs

```diff
@@ -936,15 +936,19 @@ impl IotHubClient {
                         }
                     }
                 }
-                Ok(Err(e)) => match CString::new(json!(e.to_string()).to_string()) {
-                    Ok(r) => {
-                        *response_size = r.as_bytes().len();
-                        *response = r.into_raw() as *mut u8;
-                    }
-                    Err(e) => {
-                        error!("cannot parse direct method result: {e}");
+                Ok(Err(e)) => {
+                    error!("direct method error: {e:?}");
+
+                    match CString::new(json!(e.to_string()).to_string()) {
```

Comment:
What is the difference of doing a e.to_string() vs. json!(e.to_string()).to_string() here?

Reply:
we need the error string embedded in a json doc because we wanna send it as error result content.
in ln 928 the result is already a json